File: 0387_First_Unique_Character_in_a_String/387.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        for idx in range(len(s)):
            if s.count(s[idx]) == 1:
                return idx
        return -1
a = Solution()
print(a.firstUniqChar(s = "leetcode"))
print(a.firstUniqChar(s = "loveleetcode"))
print(a.firstUniqChar(s = "aabb"))

File: 0387_First_Unique_Character_in_a_String/387.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    int firstUniqChar(string s) {
        map<char, int> appearance;
        for (char ch: s) {
            appearance[ch]++;
        }
        for (int idx = 0; idx < s.length(); idx++) {
            if (appearance[s[idx]] == 1)
                return idx;
        }
        return -1;
    }
};

int main() {
    Solution a;
    cout << a.firstUniqChar("loveleetcode") << "\n";
    return 0;
}